src/main/java/com/epam/training/array/ArrayLogic.java

```diff
@@ -1,16 +1,80 @@
 package com.epam.training.array;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class ArrayLogic {
 
-    //TODO: write it using binary search
-    public boolean exists(Array array) {
-        throw new UnsupportedOperationException();
-    }
+	public boolean exists(Array array, int element) {
+		boolean result = false;
+		int[] arraySortable = array.getItems();
+		int firstIndex = 0;
+		int lastIndex = arraySortable.length - 1;
+		while(firstIndex <= lastIndex) {
+			int middleIndex = (firstIndex + lastIndex) / 2;
+			if(arraySortable[middleIndex] == element) {
+				result = true;
+				return result;
+			}
+			if(arraySortable[middleIndex] < element) {
+				firstIndex = middleIndex + 1;
+			}
+			if (arraySortable[middleIndex] > element) {
+				lastIndex = middleIndex - 1;
+			}
+		}
+		return result;
+	}
+
+	public Array findPrimeNumbers(Array array) {
+		int[] arraySerched = array.getItems();
 
-    public Array findPrimeNumbers(Array array) {
-        throw new UnsupportedOperationException();
-    }
-    
-   //TODO: add other methods here
+		List<Integer> primeArray = new ArrayList<>();
+		for(int i = 0; i <arraySerched.length; i++) {
+			boolean isPrime = isPrimeNumber(arraySerched[i]);
+			if(isPrime) {
+				primeArray.add(arraySerched[i]);
+			}
+		}
+		int[] resultArray = new int[primeArray.size()];
+		for (int i=0 ;i < primeArray.size() ;i++) {
+			resultArray[i] = primeArray.get(i);
+		}
+		return new Array(resultArray);
+	}
+	
+	public boolean isPrimeNumber(int a) {
+		for(int i = 2; i < a; i++) {
+			if(a % i == 0) {
+				return false;
```

Comment:
try to write with one point of exit, in this case you can create local variable and return it